cluster/kubernetes/sync.go

```diff
@@ -238,10 +240,34 @@ func (c *Cluster) getResourcesBySelector(selector string) (map[string]*kuberesou
 	return result, nil
 }
 
+func (c *Cluster) listAllowedResources(
+	namespaced bool, gvr schema.GroupVersionResource, options meta_v1.ListOptions) ([]unstructured.Unstructured, error) {
+	if !namespaced || len(c.allowedNamespaces) == 0 {
+		// The resource is not namespaced or all the namespaces are allowed
+		resourceClient := c.client.dynamicClient.Resource(gvr)
+		data, err := resourceClient.List(options)
+		if err != nil {
+			return nil, err
+		}
+		return data.Items, nil
+	}
+
+	// List resources only from the allowed namespaces
+	var result []unstructured.Unstructured
+	for _, ns := range c.allowedNamespaces {
```

Comment:
Maybe I should be using `getAllowedNamespaces()` to ensure the namespaces are accessible and give a more reasonable error if they aren't?

Reply:
@squaremo thoughts?

Reply:
What I gather is that this would mean the difference between returning an error if any namespaces that are allowed happen to not exist, and simply eliding those -- yes, I think you want existing, allowed namespaces here.

(In other places you want "in principle allowed namespaces", i.e., namespaces that are allowed, whether or not they exist).

Reply:
Alright, I will change this.